src/main.rs

```diff
@@ -36,24 +36,26 @@ fn open_dit_repo() -> Result<Repository> {
 
 /// find-tree-init-hash subcommand implementation
 ///
-fn find_tree_init_hash(repo: &Repository, matches: &clap::ArgMatches) -> Result<()> {
+fn find_tree_init_hash(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // note: commit is always present since it is a required parameter
     repo.revparse_single(matches.value_of("commit").unwrap())
         .and_then(|obj| repo.find_commit(obj.id()))
         .chain_err(|| EK::WrappedGitError)
         .and_then(|commit| repo.find_tree_init(commit).chain_err(|| EK::WrappedGitDitError))
-        .map(|commit| println!("{}", commit.id()))
+        .map(|commit| {println!("{}", commit.id()); 0})
+        .unwrap_or_else(|err| {error!("{}", err); 1})
 }
 
 
 fn main() {
     let yaml    = load_yaml!("cli.yaml");
     let matches = App::from_yaml(yaml).get_matches();
 
-    let repo = open_dit_repo().unwrap();
+    let repo = open_dit_repo().unwrap(); // TODO: error handling
```

Comment:
`TODO: Fix panic` would be way more accurate :smile: 

Also, we can do a simple 

```
let repo = match open_dir_repo() { 
    Ok(r) => r, 
    Err(e) => { 
        error!("{}", e); 
        std::process::exit(1) 
    }
}
```

 here